### PR TITLE
[Android] Improvements around SendAndroidCommand to avoid ANRs.

### PR DESCRIPTION
**Pull request by jellefoks**
This makes |g_app_running| an atomic (with barriers), and ensures it's set to false before destroying ApplicationAndroid.

This should avoid dereferencing |ApplicationAndroid| during application shutdown, avoiding the corresponding crashes and ANRs.

b/225209442

### CODE REVIEW

File: starboard/android/shared/android_main.cc [hunk 1]
```diff
@@ -235,20 +237,19 @@ void* ThreadEntryPoint(void* context) {
...
+  g_app_running.store(true);
 
   // Signal GameActivity_onCreate() that it may proceed.
   g_app_created_semaphore->Put();
 
   // Enter the Starboard run loop until stopped.
-  int error_level =
-      app.Run(std::move(command_line), GetStartDeepLink().c_str());
-#endif  // SB_API_VERSION >= 15
+  error_level = app.Run(std::move(command_line), GetStartDeepLink().c_str());
```
Comment (zhongqiliang) [inline starboard/android/shared/android_main.cc:246]: why this change? any benefit to split it into 2 lines?

**Review by y4vor — APPROVED** [review]

File: starboard/android/shared/android_main.cc (hunk 1 above)
Comment (jellefoks) [reply starboard/android/shared/android_main.cc:246]: Oh, leftover from another change that I removed. I'll undo this.
